packages/application-generic/src/services/bull-mq.service.ts

```diff
@@ -114,4 +117,50 @@ export class BullmqService {
         : {}),
     });
   }
+
+  public async gracefulShutdown(): Promise<void> {
+    Logger.log('Shutting the BullMQ service down', LOG_CONTEXT);
+
+    if (this._queue) {
+      await this._queue.close();
+    }
+    if (this._worker) {
+      await this._worker.close();
+    }
+
+    Logger.log('Shutting down the BullMQ service has finished', LOG_CONTEXT);
+  }
+
+  public async getRunningStatus(): Promise<{
+    queueIsPaused: boolean;
+    workerIsRunning: boolean;
+  }> {
+    const queueIsPaused =
+      (this._queue && (await this._queue.isPaused())) || undefined;
```

Comment:
The problem is that BullMQ service is used both for creating queues and workers. We have some services that only create a queue and other that only create a worker. I wanted to do a generic functionality that showed the running status of the service that instantiates the BullMQ service. Therefore the undefined for the values to return to handle those cases. Open to a better suggestion though!

Reply:
Obviously not for this cycle but maybe we can think of splitting the logic into two classes that extend this service one producer and one consumer.

Reply:
IMO we do always call `createQueue` when initializing any queue, and the workers are tied to the queue, and won't do anything without it... so my suggestion is to move the queue creation to the constructor of the `BullMqService`, but of course we can do this in the separate ticket in the cooldown...

Reply:
I ignore the reasons why it was done this way and not like you suggest. Maybe @davidsoderberg could give more context.
Though I can infer from the PR he did that the intention was to provide flexibility to create multiple queues independently as we will only have one worker that will consume them all. If we move all the queue creation to the BullMQ service it would make harder to be able to configure every queue independently as we have implemented right now.
Also to be able to pass a `groupId` that at the end is to take advantage of BullMQ grouping feature when consuming jobs, that was one of the performance improvement actions we did in previous release.